src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/ConvertDllsToWebCil.cs

```diff
@@ -88,18 +81,14 @@ private TaskItem ConvertDll(string tmpDir, ITaskItem candidate)
 
         if (Utils.IsNewerThan(dllFilePath, finalWebcil))
         {
-            var tmpWebcil = Path.Combine(tmpDir, webcilFileName);
             var logAdapter = new LogAdapter(Log);
-            var webcilWriter = Microsoft.WebAssembly.Build.Tasks.WebcilConverter.FromPortableExecutable(inputPath: dllFilePath, outputPath: tmpWebcil, logger: logAdapter);
-            webcilWriter.ConvertToWebcil();
+            var webcilWriter = Microsoft.WebAssembly.Build.Tasks.WebcilConverter.FromPortableExecutable(logger: logAdapter, inputPath: dllFilePath);
+            var webcilData = webcilWriter.ConvertToWebcilBytes();
 
             if (!Directory.Exists(candidatePath))
                 Directory.CreateDirectory(candidatePath);
 
-            if (Utils.CopyIfDifferent(tmpWebcil, finalWebcil, useHash: true))
-                Log.LogMessage(MessageImportance.Low, $"Generated {finalWebcil} .");
-            else
-                Log.LogMessage(MessageImportance.Low, $"Skipped generating {finalWebcil} as the contents are unchanged.");
+            ArtifactWriter.PersistFileIfChanged(Log, webcilData, finalWebcil);
```

Comment:
ArtifactWriter is probably not atomic either. Running 2 of them at the same time could lead to interleaved reads and writes in random order. Am I missing something ?

Reply:
lol, I made this same comment just now. Move is what I recommend too.